src/gl.rs

```diff
@@ -601,938 +201,198 @@ pub fn tex_sub_image_3d(target: GLenum,
+    fn get_string(&self, which: GLenum) -> String;
+    fn get_shader_iv(&self, shader: GLuint, pname: GLenum) -> GLint;
+    fn compile_shader(&self, shader: GLuint);
+    fn create_program(&self) -> GLuint;
+    fn delete_program(&self, program: GLuint);
+    fn create_shader(&self, shader_type: GLenum) -> GLuint;
+    fn delete_shader(&self, shader: GLuint);
+    fn detach_shader(&self, program: GLuint, shader: GLuint);
+    fn link_program(&self, program: GLuint);
+    fn clear_color(&self, r: f32, g: f32, b: f32, a: f32);
+    fn clear(&self, buffer_mask: GLbitfield);
+    fn clear_depth(&self, depth: f64);
+    fn clear_stencil(&self, s: GLint);
+    fn flush(&self);
+    fn finish(&self);
+    fn get_error(&self) -> GLenum;
+    fn stencil_mask(&self, mask: GLuint);
+    fn stencil_mask_separate(&self, face: GLenum, mask: GLuint);
+    fn stencil_func(&self,
+                    func: GLenum,
                     ref_: GLint,
-                    mask: GLuint) {
-    unsafe {
-        ffi::StencilFunc(func, ref_, mask)
-    }
-}
-
-#[inline]
-pub fn stencil_func_separate(face: GLenum,
+                    mask: GLuint);
+    fn stencil_func_separate(&self,
+                             face: GLenum,
                              func: GLenum,
                              ref_: GLint,
-                             mask: GLuint) {
-    unsafe {
-        ffi::StencilFuncSeparate(face, func, ref_, mask)
-    }
-}
-
-#[inline]
-pub fn stencil_op(sfail: GLenum,
+                             mask: GLuint);
+    fn stencil_op(&self,
+                  sfail: GLenum,
                   dpfail: GLenum,
-                  dppass: GLenum) {
-    unsafe {
-        ffi::StencilOp(sfail, dpfail, dppass)
-    }
-}
-
-#[inline]
-pub fn stencil_op_separate(face: GLenum,
+                  dppass: GLenum);
+    fn stencil_op_separate(&self,
+                           face: GLenum,
                            sfail: GLenum,
                            dpfail: GLenum,
-                           dppass: GLenum) {
-    unsafe {
-        ffi::StencilOpSeparate(face, sfail, dpfail, dppass)
-    }
-}
-
-#[cfg(target_os="android")]
-extern {
-    pub fn glEGLImageTargetTexture2DOES(target: GLenum, image: GLeglImageOES);
-}
-
-#[cfg(target_os="android")]
-pub fn egl_image_target_texture2d_oes(target: GLenum, image: GLeglImageOES) {
-    unsafe {
-        glEGLImageTargetTexture2DOES(target, image);
-    }
+                           dppass: GLenum);
+    fn egl_image_target_texture2d_oes(&self, target: GLenum, image: GLeglImageOES);
+    fn generate_mipmap(&self, target: GLenum);
+    fn insert_event_marker_ext(&self, message: &str);
+    fn push_group_marker_ext(&self, message: &str);
+    fn pop_group_marker_ext(&self);
 }
 
 #[inline]
-pub fn generate_mipmap(target: GLenum) {
-    unsafe {
-        ffi::GenerateMipmap(target)
-    }
+pub fn buffer_data<T>(gl_: &Arc<Gl>, target: GLenum, data: &[T], usage: GLenum) {
```

Comment:
Correction: ok, you do need them outside in order to use `Gl` as a trait object. Even though I'm not sure why we require it to be a trait object, I suppose there is no harm in supporting it.

Meanwhile, these helpers should just accept `gl_: &Gl` directly instead of `gl_: &Arc<Gl>`